synapse/handlers/room_member.py

```diff
@@ -1260,10 +1260,22 @@ async def do_3pid_invite(
         if invitee:
             # Note that update_membership with an action of "invite" can raise
             # a ShadowBanError, but this was done above already.
+            # We don't check the invite against the spamchecker(s) here (through
+            # user_may_invite) because we'll do it further down the line anyway (in
+            # update_membership_locked).
```

Comment:
This is a bit confusing because at first glance we seem to do exactly this on 1271. Maybe something like

```suggestion
            # We only run the 3pid spamchecker here; the usual spamchecker 
            # methods will be invoked further down the line (in
            # update_membership_locked).
```

Reply:
> This is a bit confusing because at first glance we seem to do exactly this on 1271

Well yeah but L1271 is on a different code path so the comment make sense (i.e. we managed to turn the 3pid invite into a matrix one but we don't check it now because we already do it further down the line).

Reply:
> Well yeah but L1271 is on a different code path

Argh, so it is. Thank you.